integration-cli/docker_cli_restart_test.go

```diff
@@ -139,5 +139,27 @@ func (s *DockerSuite) TestContainerRestartwithGoodContainer(c *check.C) {
 	if MaximumRetryCount != "3" {
 		c.Fatalf("Container Maximum Retry Count is %s, expected %s", MaximumRetryCount, "3")
 	}
+}
+
+func (s *DockerDaemonSuite) TestDaemonRestartContainerWithRestartPolicyAlwaysManuallyStoppediRestart(c *check.C) {
+	c.Assert(s.d.StartWithBusybox(), check.IsNil)
+
+	out, err := s.d.Cmd("run", "-d", "--restart", "always", "busybox", "top")
+	c.Assert(err, check.IsNil)
+	id := strings.TrimSpace(out)
 
+	_, err = s.d.Cmd("stop", id)
+	c.Assert(err, check.IsNil)
+	_, err = s.d.Cmd("wait", id)
+	c.Assert(err, check.IsNil)
+
+	out, err = s.d.Cmd("ps", "-q")
+	c.Assert(err, check.IsNil)
+	c.Assert(out, check.Equals, "")
+
+	c.Assert(s.d.Restart(), check.IsNil)
+
+	out, err = s.d.Cmd("ps", "-q")
+	c.Assert(err, check.IsNil)
+	c.Assert(strings.TrimSpace(out), check.Equals, id[:12])
```

Comment:
Just reading and this may be a silly question; will this be save to test, or result in a "flaky" test? Are there situations during the test that may cause the container start to be delayed, and therefore not (yet) appearing in `ps -q`?